### PR TITLE
Add check for domain status updates in operator upgrade tests

### PR DESCRIPTION
**Pull request by vanajamukkara**
Add check for domain status updates in operator upgrade tests before and after upgrading operator from old releases to latest.

Jenkins results -
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7389/ - latest
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7357/ - full suite, failures not related to the changes in the PR

### CODE REVIEW

**Review by anpanigr — COMMENTED** [review]
How about checking other domain status such as DOMAIN_STATUS_CONDITION_FAILED_TYPE and DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE before upgrade ?

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java [hunk 1]
```diff
@@ -232,6 +238,11 @@ private void upgradeOperatorAndVerify(String externalServiceNameSuffix,
     for (int i = 1; i <= replicaCount; i++) {
       pods.put(managedServerPodNamePrefix + i, getPodCreationTime(domainNamespace, managedServerPodNamePrefix + i));
     }
+
+    // verify there is no status condition type Completed before upgrading to Latest
```
[the remark this answers is not in the document]
Comment (vanajamukkara) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java:242]: They are there in both old and new releases, no change. Completed is the new condition added.

**Review by anpanigr — APPROVED** [review]
lgtm

**Review by maggiehe00 — APPROVED** [review]
lgtm

**Pull request merged**